packages/react-app/src/pages/edit.tsx

```diff
@@ -1,166 +1,343 @@
 import { useDropzone } from "react-dropzone";
-import { useCallback, useState } from "react";
-import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
-import { DatePicker } from '@mui/x-date-pickers/DatePicker';
-import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns'
+import { useCallback, useEffect, useState, useMemo } from "react";
 import { Giveaway } from "../lib/types";
+import { GetGiveawayDetails, UpdateGiveaway } from "../lib/queryClient";
+
+import uploadIcon from './../assets/CloudUpload.svg';
+import { DatePicker, LocalizationProvider } from "@mui/x-date-pickers";
+import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFns";
+
+const textInputStyle = {
+  '& .MuiInputBase-root': {
+    backgroundColor: 'white',
+    borderRadius: '8px',
+    '& fieldset': {
+      borderColor: '#E1E6EF',
+    }
+  },
+}
+const datePickerStyle = {
+  backgroundColor: 'white',
+  borderRadius: '8px',
+  width: '100%',
+  '& .MuiInputBase-root': {
+    '& fieldset': {
+      borderColor: '#E1E6EF',
+    }
+  },
+}
+
+const fieldLabelStyle = {
+  color: '#000000'
+}
+const fieldErrorDescriptionStyle = {
+  color: '#FF0000',
+}
 
 const EditGiveaway = () => {
   const navigate = useNavigate();
   const { id } = useParams();
+  const { data } = GetGiveawayDetails(id);
+  const { handleSubmit, register, reset, control, formState, setError } = useForm<Giveaway>({
+    defaultValues: useMemo(() => data, [data])
+  });
+  useEffect(() => {
+    reset(data);
+  }, [reset, data]);
 
-  const [binFile, setBinFile] = useState<any>();
+  const [binImageFile, setBinImageFile] = useState<File>();
+  const [imageURL, setImageURL] = useState<string>('')
 
-  const onDrop = useCallback((files: any) => {
-    setBinFile(files[0])
+  const onDrop = useCallback((files: File[]) => {
+    setImageURL(URL.createObjectURL(files[0]));
+    setBinImageFile(files[0]);
   }, [])
-  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+  const { getRootProps, getInputProps } = useDropzone({
     onDrop,
     accept: {
-      'image/png': ['.png'],
+      "image/*": [".png", ".gif", ".jpeg", ".jpg"],
     },
     maxFiles: 1,
   })
 
-  const { handleSubmit, register } = useForm<Giveaway>({});
+  function addServerErrors<Giveaway>(
+    errors: { [P in keyof Giveaway]?: string },
+    setError: (
+      fieldName: keyof Giveaway,
+      error: { type: string; message: string }
+    ) => void
+  ) {
+    return Object.keys(errors).forEach((key) => {
+      const field = key as keyof Giveaway;
+      if (field !== undefined) {
+        setError(field, {
+          type: "server",
+          message: errors[field] as string
+        });
+      }
+    });
+  }
+
+  const savePress = async (data: FormData) => {
+    const result = await UpdateGiveaway(data)
+    if (!result.success && result.errors) {
+      addServerErrors(result.errors, setError);
+    } else {
+      navigate(-1);
```

Comment:
Navigating back might give the wrong idea that it was correctly uploaded. I think in this case we should prompt an error message. Maybe with https://mui.com/material-ui/react-snackbar/ as we have on login page